modules/core/domain/src/main/java/org/rhq/core/domain/content/PackageVersion.java

```diff
@@ -139,7 +142,13 @@
         + "   AND pv.repoPackageVersions IS EMPTY " //
         + "   AND pv.installedPackages IS EMPTY " //
         + "   AND pv.installedPackageHistory IS EMPTY "),
-
+    @NamedQuery(name = PackageVersion.DELETE_BY_PKG_IF_NO_CONTENT_SOURCES_OR_REPOS, query = "DELETE PackageVersion pv "
+        + " WHERE pv.id NOT IN (SELECT pvcs.packageVersion.id " //
+        + "                       FROM PackageVersionContentSource pvcs) " //
```

Comment:
Perhaps IN clause could be avoided by replacing with:
  NOT EXISTS ( SELECT * FROM PackageVersionContentSource pvcs WHERE pvcs.packageVersion.id = pv.id )